awx/main/dispatch/reaper.py

```diff
@@ -0,0 +1,42 @@
+from datetime import timedelta
+import logging
+
+from django.db.models import Q
+from django.utils.timezone import now as tz_now
+from django.contrib.contenttypes.models import ContentType
+
+from awx.main.models import Instance, UnifiedJob, WorkflowJob
+
+logger = logging.getLogger('awx.main.dispatch')
+
+
+def reap(instance=None, status='failed'):
```

Comment:
I extracted the job reaping code into a distinct function that we can test here.  It is *no longer* called in the periodic task manager.  Instead, it's called in two locations:

1.  When a dispatcher starts, it reaps any running jobs that match *its* execution node.  In this way, if the dispatcher restarts (or the node reboots), it cleans up after itself when it comes back online.
2.  If a node _never_ comes back online, other nodes in a cluster will notice and set the capacity to zero.  With the changes in _this_ PR, that process *also* calls this reaping function to mark jobs as failed for nodes that go missing in a cluster.

Reply:
Yea.  If there's a single node and the node goes down, there _is_ no watcher.  But that was the case before w/ the task manager right?  The task manager will never run if there's no celery/dispatcher running anywhere.

Reply:
Correct, figure it was just worth stating for completeness.